packages/openapi-generator/src/parser/unique-naming.ts

```diff
@@ -74,3 +125,81 @@ function isDuplicate(name: string, correctNames: string[]): boolean {
 function isFormatted(name: string, format: (name: string) => string): boolean {
   return format(name) === name;
 }
+
+function hasDuplicates(names: string[]): boolean {
+  const uniqueNames = unique(names);
+  return uniqueNames.length !== names.length;
+}
+
+function hasReservedWords(names: string[], reservedWords: string[]): boolean {
+  return !!reservedWords.find(reservedWord => names.includes(reservedWord));
+}
+
+function getDuplicateErrorMessage(
+  names: string[],
+  formattedNames: string[],
+  reservedWords: string[]
+): string {
+  const uniqueNames = unique(formattedNames);
+
+  let errorMessage = '';
+
+  if (hasDuplicates(formattedNames)) {
```

Comment:
It seems, when reaching this line, it is already checked in the validation step before. Not sure, whether for a large spec, whether this will be some scalability issue.
Same for the `hasReservedWords` below.
If we want to improve, we can return some information about e.g., "duplicate names" during the validation, which should be a relatively small data set.

Reply:
Yes, I also didn't like this too much. I did it similarly to the way you suggested initially, but I changed it, because I think it is better to separate the validation and error handling. 
I will try to find a way to implement a nicer "result" for the validation...

Reply:
Argh, I just tried something and I struggle to find a way that is still nicely readable while not getting the unique values multiple times.

I also did the reserved words reduction before, but there I have to go over all reserved words once anyways to find those that are reserved, so I wouldn't really gain much from this.

I think it is just not worth to sacrifice the readability here, especially because this will only be executed in the error case. Of course, feel free to correct me and provide an example, maybe you have a better way in mind than me.